### PR TITLE
Registration modification of transferring process.

### PR DESCRIPTION
**Pull request by kithminisg**

### CODE REVIEW

**Review by Udith-Gayan — APPROVED** [review]

File: src/evernodeone.c [hunk 1]
```diff
@@ -77,7 +97,10 @@ int64_t hook(uint32_t reserved)
             TRACEVAR(host_reg_fee);
 
             if (float_compare(float_amt, float_set(0, host_reg_fee), COMPARE_LESS) == 1)
-                rollback(SBUF("Evernode: Amount sent is less than the minimum fee for host registration."), 1);
+            {
+                if ((has_initiated_transfer != 0) && (float_compare(float_amt, float_set(0, NOW_IN_EVRS), COMPARE_LESS) == 1))
```
Comment (chalith) [inline src/evernodeone.c:101]: Conditional check? `(has_initiated_transfer != 0) ? () : ()`

File: src/evernodeone.c [hunk 2]
```diff
@@ -87,10 +110,6 @@ int64_t hook(uint32_t reserved)
             uint8_t host_addr[HOST_ADDR_VAL_SIZE];
             CLEAR_BUF(host_addr, 0, HOST_ADDR_VAL_SIZE); // Initialize buffer wih 0s
 
-            // <host_address(20)><cpu_model_name(40)><cpu_count(2)><cpu_speed(2)><cpu_microsec(4)><ram_mb(4)><disk_mb(4)>
-            uint8_t token_id[TOKEN_ID_VAL_SIZE];
-            CLEAR_BUF(token_id, 0, TOKEN_ID_VAL_SIZE); // Initialize buffer wih 0s
-
             if (state(SBUF(host_addr), SBUF(STP_HOST_ADDR)) != DOESNT_EXIST)
                 rollback(SBUF("Evernode: Host already registered."), 1);
```
Comment (chalith) [inline src/evernodeone.c:115]: Can't we move "Generate the NFT token id." to `if (has_initiated_transfer == 0)` block

File: src/evernodeone.c [hunk 3]
```diff
@@ -67,6 +67,26 @@ int64_t hook(uint32_t reserved)
             int is_evr;
             IS_EVR(is_evr, amount_buffer, issuer_accid);
 
+            // Checking whether this host has an initiated transfer to continue.
```
Comment (chalith) [inline src/evernodeone.c:70]: What if we move this block after `is_evr` check

File: src/evernodeone.c [hunk 4]
```diff
@@ -67,6 +67,26 @@ int64_t hook(uint32_t reserved)
...
+            TRANSFEREE_ADDR_KEY(account_field);
+
+            uint8_t transferee_addr[TRANSFEREE_ADDR_VAL_SIZE];
+            uint8_t prev_host_addr[HOST_ADDR_VAL_SIZE];
+            uint8_t prev_token_id[TOKEN_ID_VAL_SIZE];
+            if (state(SBUF(transferee_addr), SBUF(STP_TRANSFEREE_ADDR)) != DOESNT_EXIST)
+            {
+                has_initiated_transfer = 1;
+
+                HOST_ADDR_KEY((uint8_t *)(transferee_addr + TRANSFER_HOST_ADDRESS_OFFSET));
```
Comment (chalith) [inline src/evernodeone.c:81]: Feels like more readable if we move this `HOST_ADDR_KEY` and `TOKEN_ID_KEY` to later else block whether we check `has_initiated_transfer == 0`

File: src/evernodeone.c [hunk 5]
```diff
@@ -212,100 +228,158 @@ int64_t hook(uint32_t reserved)
...
+                COPY_BUF(prev_token_id, HOST_ADDRESS_OFFSET, account_field, 0, ACCOUNT_ID_SIZE);
+                COPY_BUF_NON_CONST_LEN_GUARDM(prev_token_id, HOST_CPU_MODEL_NAME_OFFSET, cpu_model_ptr, 0, cpu_model_len, CPU_MODEl_NAME_LEN, 1, 1);
+                UINT16_TO_BUF(&prev_token_id[HOST_CPU_COUNT_OFFSET], cpu_count);
+                UINT16_TO_BUF(&prev_token_id[HOST_CPU_SPEED_OFFSET], cpu_speed);
+                UINT32_TO_BUF(&prev_token_id[HOST_CPU_MICROSEC_OFFSET], cpu_microsec);
+                UINT32_TO_BUF(&prev_token_id[HOST_RAM_MB_OFFSET], ram_mb);
+                UINT32_TO_BUF(&prev_token_id[HOST_DISK_MB_OFFSET], disk_mb);
+
+                // Set the STP_TOKEN_ID correctly.
+                TOKEN_ID_KEY((uint8_t *)(prev_host_addr + HOST_TOKEN_ID_OFFSET));
```
Comment (chalith) [inline src/evernodeone.c:358]: This can be removed with [this comment](https://github.com/EvernodeXRPL/evernode-hook/pull/92/files#r1035584968)

**Review by chalith — APPROVED** [review]

**Pull request merged**